### PR TITLE
virttest.test_setup.HugePageConfig:Skip test when host environmnet does not meet test requirement

### PR DESCRIPTION
**Pull request by yanan-fu**
Return "skip"  when host environmnet does not meet test requirement.

Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
Checking  "add_mem" function libvirt_vm.py now.

File: virttest/test_setup.py [hunk 1]
```diff
@@ -367,8 +360,8 @@ def check_hugepage_size_as_expected(self):
         """
         error_context.context("Check whether the hugepage size as expected")
         if self.hugepage_size != self.expected_hugepage_size:
-            raise HPNotSupportedError("The current hugepage size on host does"
-                                      "not match the expected hugepage size.\n")
+            self.test.cancel("The current hugepage size on host does"
```
Comment (balamuruhans) [inline virttest/test_setup.py]: It could be more appropriate with TestSetupError(),

File: virttest/test_setup.py [hunk 2]
```diff
@@ -350,15 +343,15 @@ def check_hugepage_support(self):
...
+            self.test.cancel("Your host does not support hugepage, "
+                             "as miss the cpu flag %s on your host."
+                             "Please check cpu flags %s on the host" %
+                             (self.hugepage_cpu_flag, host_cpu_flags))
         if self.hugepage_match_str not in host_ker_cml:
-            raise HPNotSupportedError("Your host does not support hugepage, "
-                                      "as miss the %s in host kernel cmdline."
-                                      "Please check kernel cmdline %s on host" %
-                                      (self.hugepage_match_str, host_ker_cml))
+            self.test.cancel("Your host does not support hugepage, "
```
Comment (balamuruhans) [inline virttest/test_setup.py]: can we use exceptions.TestSkipError() from framework so that we can avoid test here?

**Comment by yanan-fu** [comment]
@balamuruhans  drop the previous method, and use TestSkipError instead. Could you help review again, thanks a lot!

**Comment by yanan-fu** [comment]
@CongLi  could you help review this patch, thanks a lot !

**Comment by jingzhao84** [comment]
LGTM, ACKed: jinzhao@redhat.com

(1/4) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.2M: PASS (602.95 s)
 (2/4) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.16M: SKIP
 (3/4) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.512M: SKIP
 (4/4) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.boot_vm_in_hugepage.1G: SKIP

**Comment by luckyh** [comment]
@waynesun09 @kylazhang would you please help review this pr? Thanks!

File: virttest/test_setup.py [hunk 3]
```diff
@@ -66,14 +66,6 @@ class THPKhugepagedError(THPError):
     pass
 
 
-class HPNotSupportedError(Exception):
```
Comment (CongLi) [inline virttest/test_setup.py]: I prefer to keep class HPNotSupportedError() although there is no other place to call it,
but it's still useful.

Comment (yanan-fu) [reply virttest/test_setup.py]: If no other place call it, why keep it as a dead code ?  could you explain more about its use ?

**Comment by balamuruhans** [comment]
Let us merge this. Thanks @yanan-fu and all for review.

**Pull request merged**